WordPress/wordcamp.org
# Fonts: Update editor font handling, load custom fonts for WordCamp Europe

This PR does a few things to the way fonts are handled in the editor:

1. Ignore the fonts folder which is synced down from SVN, we don't need to commit it in github too (that folder will go away with GB 17.6+)
2. Remove the Font API code which is currently throwing "Deprecated" warnings, it's not working as it should anyway
3. Add in a workaround to load custom fonts for WordCamp Europe
4. Re-enable loading the google fonts (the old way) on block sites - this will be a quick workaround to get WCUS's 2023 site working again.

Fixes #1203, See #1075.

The new code is in this one commit, that might be easier to review: https://github.com/WordPress/wordcamp.org/commit/7b9285676630533609e2a7cabb2fbb091dc020bf

Once GB 17.6 is released, we'll test that it works correctly on the WordCamp network, then remove the workaround. The user styles (what's found in the `wp_global_styles` post type) will probably need to be updated with the new file sources.

### Screenshots

Krona One and Lora now show up in the Font Library as "theme fonts."

<img width="1015" alt="Screenshot 2024-01-19 at 4 21 15 PM" src="https://github.com/WordPress/wordcamp.org/assets/541093/fb9706dc-7179-451c-997b-642a3901a6cf">

### How to test the changes in this Pull Request:

1. The fonts are coded to only load on WC Europe and the test site, so if you're testing locally, update the IDs in `inject_fonts_theme_json`.
2. Use a block theme (2023 or 2024)
3. View the font library in the site editor, the two fonts should be available in "Theme Fonts" - make sure they're active
5. Edit a post or page, the fonts should be available in the Font Family dropdown
6. Update some text to use the new fonts, it should load the fonts in the editor
7. View the frontend, the fonts should be loading correctly
